### ISSUE TITLE
[Mollie Payments Plugin] Order line amounts are not refunded only shipping lines

### ISSUE THREAD

**Issue by seminarian** [vendure-ecommerce/vendure#2338]
**Describe the bug**
When doing a refund via Admin UI the `amount` sent to Mollie is not correct. Order line costs are not sent. Only shipping cost.

So there are two scenarios where things get dangerous:
1. No shipping cost gets refunded => refunding fails with message in Admin UI though the Order is cancelled. (Fails because a refund with amount 0 is sent to Mollie).
![image](https://github.com/vendure-ecommerce/vendure/assets/9881519/0e907064-6eea-4246-ba35-02914229617e)
```
[dev:server] {
[dev:server]   paymentId: 'tr_EurVpHQqdc',
[dev:server]   description: 'Customer request',
[dev:server]   amount: { value: '0.00', currency: 'EUR' }
[dev:server] }
```
2.  shipping cost + item lines are selected for refund => Refund seems to work but the amount is not correct
Only the shipping cost amount is refunded.
```
[dev:server] {
[dev:server]   paymentId: 'tr_g7BcVK8KZ6',
[dev:server]   description: 'Customer request',
[dev:server]   amount: { value: '3.99', currency: 'EUR' }
[dev:server] }
```

**To Reproduce**
1. Go to the Order you would like to refund
2. Click on Cancel order
4. Refund all except the shipping cost
5. See error popping up in Admin UI and no "refunds" are added

**Environment (please complete the following information):**
 - @vendure/core version: 2.0.3
 - Nodejs version: v16.19.0
 - Database (mysql/postgres etc): postgres

@martijnvdbrug Would you have this issue as well?

**Comment by seminarian** [comment]
@michaelbromley after thinking a bit more about it this might be something going wrong in the Vendure Core actually and not in the payment plugin?

**Comment by martijnvdbrug** [comment]
Hi @seminarian,

Sounds like a big deal. However, I'm having some trouble reproducing your issue: Both canceling as well as refunding seem to work with the dev-server:

![image](https://github.com/vendure-ecommerce/vendure/assets/6604455/951692f7-a855-439b-a05e-fe3218f8a79f)

![image](https://github.com/vendure-ecommerce/vendure/assets/6604455/428558a1-5704-4d30-bd6c-dc43d0c007b8)

![image](https://github.com/vendure-ecommerce/vendure/assets/6604455/a305f3f3-dbae-4da6-9754-ae680d247a7a)


Is there any way you can reproduce this with the Mollie dev-server?

1. Checkout branch [`martijnvdbrug:fix/mollie-dev-server`](https://github.com/martijnvdbrug/vendure/tree/fix/mollie-dev-server)(It contains some fixes to the Mollie dev-server)
2. `yarn`, then `yarn bootstrap`, then `yarn build` in the root of the project
3. Create the file `packages/payments-plugin/.env` with content `MOLLIE_APIKEY=your-test-apikey`
3. `cd packages/payments-plugin`
4. `yarn dev-server:mollie`
6. Click the link in your console:
![image](https://github.com/vendure-ecommerce/vendure/assets/6604455/744b12f1-4abd-4769-b3f2-29f75f7ec276)
7. Pay via your Mollie test account. (Redirect won't work, but the order payment is settled)
8. Go to http:localhost:3050/admin and login with superadmin/superadmin
9. Cancel and refund should work. You can verify in your Mollie dashboard if the payment has indeed been refunded.

Let me know if I can assist you with anything

**Comment by seminarian** [comment]
@martijnvdbrug 
I tried reproducing in a new project. Reproduction worked on v2.0.3 BUT then I tried with v2.0.5 and reproducing failed.
So seems to be already fixed in 2.0.5!

Apologies for your spent time and effort. I'm closing this issue.
